3party/minizip/CMakeLists.txt

```diff
@@ -2,6 +2,10 @@ project(minizip)
 
 add_definitions(-DUSE_FILE32ZPI -DNOCRYPT)
 
+if (PLATFORM_ANDROID AND (${OS} MATCHES "mac"))
+  add_definitions(-D__APPLE__)
```

Comment:
Are you sure? Why do we need to define __APPLE__ on ANDROID platform?

Reply:
Android ndk built-in cmake have some problems with platform detection. Error is happening here: 3party/minizip/ioapi.c:17

```
#if defined(__APPLE__) || defined(IOAPI_NO_64)
// In darwin and perhaps other BSD variants off_t is a 64 bit value, hence no need for specific 64 bit functions
#define FOPEN_FUNC(filename, mode) fopen(filename, mode)
#define FTELLO_FUNC(stream) ftello(stream)
#define FSEEKO_FUNC(stream, offset, origin) fseeko(stream, offset, origin)
#else
#define FOPEN_FUNC(filename, mode) fopen64(filename, mode)
#define FTELLO_FUNC(stream) ftello64(stream)
#define FSEEKO_FUNC(stream, offset, origin) fseeko64(stream, offset, origin)
#endif
```

Android ndk cmake don't switch __APPLE___ flag ON, and compiler try to use fopen64, ftello64, fseeko64 which are not defined in darwin platform.

Summary: This code block is just "kostyl" to build android in Mac correctly.